### DISCUSSION TITLE
Strange/uneven distribution of traffic based on source?

### DISCUSSION THREAD

**Discussion by bdruth** [Ysurac/openmptcprouter#1729] [Q&A]
(reposting from IRC, where it looks like it was missed)

I did some testing tonight and found a very strange (reproducible) behavior ... if I test from a computer on my LAN, I get very little usage of my wan2 connection, relative to the wan1 connection (wan1 is master). However, when I run the exact same test from the OMR box, I get very good utilization of both connections. I'm using iperf3, leveraging omr-iperf. This is reflected on both upload & download. Here's a chart from grafana using data from collectd running on the OMR box: ![grafana](https://i.imgur.com/bGqXtbJl.png)
you can see 3 runs, from OMR, from a LAN box, from OMR again (to verify). My iperf3 command options were -P 1 -R -t 60 (download) or -P 1 -t 60 (upload). Now, technically my "LAN" side is behind another router (eero), so I can't say for certain that it's an issue with the LAN interface, per se, but I will try to bypass the eero tomorrow and see what I can determine. Behind the eero, it is all wired via gigabit switches on my LAN, so it's not using WiFi or anything.

Any thoughts on what might be going on here?

**Comment by bdruth** [comment]
I haven't had a chance to conduct a test isolated from the eero, but when I switched which `wan` interface is MPTCP master, it switched which `wan` interface picked up the traffic when doing a test from the `LAN` side. Traffic was split as expected when testing from the OMR box.

**Reply by Ysurac** [reply]
What is the OMR release tested ? Where is located the iperf server side ?

**Reply by bdruth** [reply]
OMR v0.57.3 r0+15225-bfc433efd4, iperf is the one running on VPS, installed by the OMR-vps script - I think I'm on omr-vps 0.992.

**Reply by bdruth** [reply]
Any other thoughts @Ysurac ?

**Reply by Ysurac** [reply]
0.992 is a really old release. You can check using ```cat /etc/motd``` on the VPS the version. If it's really this release, update.
Make also sure that no connection have IPv6 or DHCP enabled.

**Reply by bdruth** [reply]
I do have DHCP on the Starlink connection because the dish uses CGNAT and doesn't provide a fixed IP of any kind - I'd have to put the Starlink router in-line to get a stable private, non-CGNAT addr. What is that impacting?

**Reply by bdruth** [reply]
Looks like it's `OpenMPTCProuter VPS 0.1024` - I'll run the update just to be sure there isn't a newer one.
